### PR TITLE
 feat: revamp code snippets tutorial

### PR DESCRIPTION
**Pull request by p7nov**
Rewrite the tutorial [Run code snippets](https://kotlinlang.org/docs/run-code-snippets.html) to show actual options: scratches and worksheets in the IDE, online playground in browser, ki shell in the command line.

### CODE REVIEW

**Comment by p7nov** [comment]
[WIP]: figuring out a better place and name for this page. Content is ready for review.

**Comment by p7nov** [comment]
@dromanov-jb, @ligee could you check the parts about the playground and ki-shell correspondingly?

File: docs/topics/run-code-snippets.md
Comment (etolstoy) [inline docs/topics/run-code-snippets.md:11]: AFAIK AndroidStudio also supports scratches. Shouldn’t we mention it?

File: docs/topics/run-code-snippets.md
Comment (etolstoy) [inline docs/topics/run-code-snippets.md:60]: @yalishevant Do you remember the results of our discussions about «Run as REPL» feature? Do we plan to deprecate it?

**Review by etolstoy — APPROVED** [review]

File: docs/topics/run-code-snippets.md
Comment (p7nov) [reply docs/topics/run-code-snippets.md:11]: Yeah, it does. Let's change this to `IDE` and mention Android Studio in this section.

**Comment by ligee** [comment]
About the `ki-shell` example session snapshots:
first, the latest version (0.4.5) fixed Slf4j warnings, so the output is a bit cleaner.
and second - the version of ki contains `SNAPSHOT` substring, that doesn't seem like a good signal to me. I asked maintainers to remove it in the next release.
So, it makes sense to wait until the next release and remake snapshots.

File: docs/topics/run-code-snippets.md
Comment (ligee) [inline docs/topics/run-code-snippets.md:181]: I'm not sure if we want to advertise the plugin authoring yet. I would skip this section and refer to the ki repository for further details (without mentioning extensibility here at all).

Comment (p7nov) [reply docs/topics/run-code-snippets.md]: Thanks. Removed the part about extensibility and added the GitHub link explicitly in the intro.

**Comment by p7nov** [comment]
[WIP] again: waiting for the new release of ki shell with clearer output to make screenshots.

**Comment by p7nov** [comment]
> About the `ki-shell` example session snapshots: first, the latest version (0.4.5) fixed Slf4j warnings, so the output is a bit cleaner. and second - the version of ki contains `SNAPSHOT` substring, that doesn't seem like a good signal to me. I asked maintainers to remove it in the next release. So, it makes sense to wait until the next release and remake snapshots.

@ligee I've made new screenshots on ki shell 0.4.5 ( without `-SNAPSHOT` and slf4j warnings)

File: docs/topics/run-code-snippets.md
Comment (yalishevant) [reply docs/topics/run-code-snippets.md:60]: Yes, we _decided_ to remove this check-box. Run as REPL will be the only way of running.
We heaven't planned it for a particular time and person who will do this. Roma Golyshev is now estimating the scope of work.

Comment (p7nov) [reply docs/topics/run-code-snippets.md:60]: So we're leaving it as is, I guess? I'll keep in mind the future update. Could you share the corresponding YT issue or add me to watchers?

**Review by ligee — APPROVED** [review]
Looks great to me!

File: docs/topics/run-code-snippets.md
Comment (koshachy) [reply docs/topics/run-code-snippets.md:11]: Not a mistake, just trying to find a better visual representation for a text.

File: docs/topics/run-code-snippets.md
Comment (koshachy) [inline docs/topics/run-code-snippets.md:75]:
```suggestion
* write the executable part in the body of the `main()` function.
```

File: docs/topics/run-code-snippets.md
Comment (koshachy) [inline docs/topics/run-code-snippets.md:88]: Link to the EAP? 
Or it could just distract user? 🤔

**Review by koshachy — APPROVED** [review]
Nice job.
Left some comments.

File: docs/topics/run-code-snippets.md
Comment (p7nov) [reply docs/topics/run-code-snippets.md:88]: Yeah, why not. Thanks for the idea

File: docs/topics/run-code-snippets.md
Comment (p7nov) [reply docs/topics/run-code-snippets.md:151]: Yes, let's keep it shorter

File: docs/topics/run-code-snippets.md
Comment (p7nov) [reply docs/topics/run-code-snippets.md:11]: Arranged the definitions in a list

**Pull request merged**
